libfreerdp/core/update.c

```diff
@@ -427,37 +466,74 @@ BOOL update_read_pointer_color(wStream* s, POINTER_COLOR_UPDATE* pointer_color,
 		Stream_Seek_UINT8(s); /* pad (1 byte) */
 
 	return TRUE;
+fail:
+	return FALSE;
+}
+
+POINTER_COLOR_UPDATE* update_read_pointer_color(rdpUpdate* update, wStream* s, BYTE xorBpp)
+{
+	POINTER_COLOR_UPDATE* pointer_color = calloc(1, sizeof(POINTER_COLOR_UPDATE));
+
+	if (!pointer_color)
+		goto fail;
+
+	if (!_update_read_pointer_color(s, pointer_color, xorBpp))
+		goto fail;
+
+	return pointer_color;
+fail:
+	free_pointer_color_update(update->context, pointer_color);
+	return NULL;
 }
 
-BOOL update_read_pointer_new(wStream* s, POINTER_NEW_UPDATE* pointer_new)
+POINTER_NEW_UPDATE* update_read_pointer_new(rdpUpdate* update, wStream* s)
 {
+	POINTER_NEW_UPDATE* pointer_new = calloc(1, sizeof(POINTER_NEW_UPDATE));
+
+	if (!pointer_new)
+		goto fail;
+
 	if (Stream_GetRemainingLength(s) < 2)
-		return FALSE;
+		goto fail;
 
 	Stream_Read_UINT16(s, pointer_new->xorBpp); /* xorBpp (2 bytes) */
 
 	if ((pointer_new->xorBpp < 1) || (pointer_new->xorBpp > 32))
 	{
 		WLog_ERR(TAG,  "invalid xorBpp %"PRIu32"", pointer_new->xorBpp);
-		return FALSE;
+		goto fail;
 	}
 
-	return update_read_pointer_color(s, &pointer_new->colorPtrAttr,
-	                                 pointer_new->xorBpp); /* colorPtrAttr */
+	if (!_update_read_pointer_color(s, &pointer_new->colorPtrAttr,
```

Comment:
Should be `_update_read_pointer_new`

Reply:
No, the data contained is a color poiner.

Reply:
Yes - you'r right 👍